Lesson_04/homework_04.py

```diff
@@ -0,0 +1,130 @@
+letter_h = new_adventures_of_tom_sayer.count('h')
+_log.info('Letter "h" is found %s times in the text.', letter_h)
+
+# task 05
+# Виведіть, скільки слів у тексті починається з Великої літери?
+# Replacing " with empty spaces to prevent including " in words.
+# Creating list of words from the string with split function.
+words = new_adventures_of_tom_sayer.replace('"', '').split()
+# Creating new list of words with capitalized letter.
+# Calculate quantity of words with capitalized letter with function len().
+uppercase_words = len([word for word in words if word.istitle()])
+_log.info('There are %s title letter words in the text.', uppercase_words)
+
+# task 06
+# Виведіть позицію, на якій слово Tom зустрічається вдруге
+# Find first position of word 'Tom' in the text
+first_position = new_adventures_of_tom_sayer.find('Tom')
+# Find and display second position of word 'Tom' in the text.
+second_position = new_adventures_of_tom_sayer.find('Tom', first_position+1)
+output = (f'The word "Tom" is located in {second_position} position'
+          f' for the second time in the text.')
+_log.info(output)
+
+# task 07
+# Розділіть змінну new_adventures_of_tom_sayer по кінцю речення.
+# Збережіть результат у змінній adventures_of_tom_sayer_sentences
+# Splitting variable new_adventures_of_tom_sayer by sentence end.
+adventures_of_tom_sayer_sentences = new_adventures_of_tom_sayer.split('. ')
+_log.info(adventures_of_tom_sayer_sentences)
+# Calculating quantity of sentences in the adventures_of_tom_sayer_sentences.
+sentences = f'There are {len(adventures_of_tom_sayer_sentences)} sentences.'
+_log.info(sentences)
+
+# task 08
+# Виведіть четверте речення з adventures_of_tom_sayer_sentences.
+# Перетворіть рядок у нижній регістр.
+_log.info(adventures_of_tom_sayer_sentences[3])  # Display forth sentence.
+# Modifying sentence where all word are lower  case.
+sentence_lower = adventures_of_tom_sayer_sentences[3].lower()
+_log.info(sentence_lower)
+
+# task 09
+# Перевірте чи починається якесь речення з "By the time".
+# Check if any sentence in list adventures_of_tom_sayer_sentences
+# starts with 'By the time'.
+# Result: True or False.
+# Also we need to know exactly sentence quantity in the ist.
+sentence1 = adventures_of_tom_sayer_sentences[0].startswith('By the time')
+sentence2 = adventures_of_tom_sayer_sentences[1].startswith('By the time')
+sentence3 = adventures_of_tom_sayer_sentences[2].startswith('By the time')
+sentence4 = adventures_of_tom_sayer_sentences[3].startswith('By the time')
+sentence5 = adventures_of_tom_sayer_sentences[4].startswith('By the time')
+
+# Option to find and display which sentence begins with 'By the time'
+# in the list adventures_of_tom_sayer_sentences.
+# If we don't exactly sentences in list and we want to display those sentence
+# which begins with 'By the time', this is best option.
+_log.info([sentence for sentence in adventures_of_tom_sayer_sentences if
```

Comment:
I think here could be used smth like this
`print(any(item.startswith('By the time') for item in adventures_of_tom_sayer.split('\n')))`
But agree there was no such or list comprehension info during past four lessons